### DISCUSSION TITLE
Block all topics except those I have defined

### DISCUSSION THREAD

**Discussion by gvd22** [NVIDIA-NeMo/Guardrails#344] [Q&A] [answered]
Is it possible to block all topics except those I have defined?

define user ask llm
        "what is the llama 2 model?"
        "tell me about Meta's new LLM"
        "what are the differences between Falcon and Llama"

define flow llm
    user ask llm
    bot answer

define bot answer blocked
    "I only answer questions about the Llama 2 Model.  Please ask me about that."

define flow fallback
  user ask *
  bot answer blocked

**Comment by drazvan** [comment, marked as answer]
HI @gvd22 !

This is related to https://github.com/NVIDIA/NeMo-Guardrails/discussions/129. 

Currently, there's no explicit way to block "all other topics". Right now, you could go with the approach where you also define some other topics and teach the LLM to ignore them (like here: https://github.com/NVIDIA/NeMo-Guardrails/blob/develop/examples/bots/abc/rails/disallowed.co), or you explicitly define an off-topic intent and try to add representative examples. 

We are researching better ways to enable the definition of "just the topic of interest." We'll keep you posted.

**Reply by gvd22** [reply]
Thanks for the answer :)
